smart-employee-agent/agent/main.py

```
@@ -0,0 +1,553 @@
+
+        system_prompt = build_system_prompt(session)
+        messages = [{"role": "system", "content": system_prompt}]
+        # Replay only the most recent N turns to bound latency/cost and avoid
+        # blowing past the model's context window.
+        history_window = session.chat_history[-(MAX_CHAT_HISTORY_TURNS * 2):]
+        messages.extend(history_window)
+        messages.append({"role": "user", "content": user_message})
+
+        response = await agent.ainvoke({"messages": messages})
+
+    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)